tests/unit/pmodel/test_optimal_chi.py

```diff
@@ -0,0 +1,182 @@
+"""Testing Optimal submodule."""
+
+
+import numpy as np
+import pytest
+
+from pyrealm.pmodel.optimal_chi import (
+    OptimalChiC4,
+    OptimalChiC4NoGamma,
+    OptimalChiC4NoGammaRootzoneStress,
+    OptimalChiC4RootzoneStress,
+    OptimalChiLavergne20C3,
+    OptimalChiLavergne20C4,
+    OptimalChiPrentice14,
+    OptimalChiPrentice14RootzoneStress,
+)
+from pyrealm.pmodel.pmodel_environment import PModelEnvironment
+
+
+@pytest.fixture
+def photo_env():
+    """Photosynthesis Environment setup."""
+    return PModelEnvironment(
+        tc=np.array([20]),
+        vpd=np.array([1000]),
+        co2=np.array([400]),
+        patm=np.array([101325.0]),
+        rootzonestress=np.array([1]),
+        theta=np.array([0.5]),
+    )
+
+
+@pytest.mark.parametrize(
+    """optimal_chi_class""",
+    [
+        OptimalChiPrentice14,
+        OptimalChiPrentice14RootzoneStress,
+        OptimalChiC4NoGammaRootzoneStress,
+        OptimalChiC4,
+        OptimalChiC4RootzoneStress,
+        OptimalChiLavergne20C3,
+        OptimalChiLavergne20C4,
+        OptimalChiC4NoGamma,
+    ],
+)
+def test_set_beta(optimal_chi_class, photo_env):
+    """Test that beta is set correctly."""
+    optimal_chi_instance = optimal_chi_class(env=photo_env)
+    optimal_chi_instance.set_beta()
```

Comment:
Importing within the test would need something like using strings in the parameterisation and then:

```python
import pyrealm.pmodel.optimal_chi as oc_mod
chi_class = getattr(oc_mod, optimal_chi_class)
```

Reply:
Put it in a new PR.